clang/lib/AST/ExprConstant.cpp

```diff
@@ -9325,6 +9325,9 @@ bool PointerExprEvaluator::VisitCastExpr(const CastExpr *E) {
       Result.IsNullPtr = false;
       return true;
     } else {
+      if (!Value.isLValue())
```

Comment:
I think it's worth adding a comment here explaining under what circumstances this wouldn't be an lvalue. Something along the lines of "In rare instances, the value isn't an lvalue. For example, when the value is the difference between the address of two labels. We reject that as a constant expression because we can't compute a valid offset to convert into a pointer."

Reply:
@AaronBallman Thanks for the feedback 👍. I've added all the requested changes.